packages/fx-core/src/question/templates.ts

```diff
@@ -0,0 +1,460 @@
+  {
+    id: "message-extension-ts",
+    name: TemplateNames.MessageExtension,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "message-extension-js",
+    name: TemplateNames.MessageExtension,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "m365-message-extension-ts",
+    name: TemplateNames.MessageExtensionM365,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "m365-message-extension-js",
+    name: TemplateNames.MessageExtensionM365,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "message-extension-action-ts",
+    name: TemplateNames.MessageExtensionAction,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "message-extension-action-js",
+    name: TemplateNames.MessageExtensionAction,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "message-extension-copilot-ts",
+    name: TemplateNames.MessageExtensionCopilot,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "message-extension-copilot-js",
+    name: TemplateNames.MessageExtensionCopilot,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "link-unfurling-ts",
+    name: TemplateNames.LinkUnfurling,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "link-unfurling-js",
+    name: TemplateNames.LinkUnfurling,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "api-plugin-existing-api",
+    name: TemplateNames.ApiPluginExistingApi,
+    language: "common",
+    description: "",
+  },
+];
+
+const copilotPluginTemplates: Template[] = [
+  {
+    id: "copilot-plugin-from-scratch-ts",
+    name: TemplateNames.CopilotPluginFromScratch,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "copilot-plugin-from-scratch-js",
+    name: TemplateNames.CopilotPluginFromScratch,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "copilot-plugin-from-scratch-api-key-ts",
+    name: TemplateNames.CopilotPluginFromScratchApiKey,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "copilot-plugin-from-scratch-api-key-js",
+    name: TemplateNames.CopilotPluginFromScratchApiKey,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "api-plugin-from-scratch-js",
+    name: TemplateNames.ApiPluginFromScratch,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "api-plugin-from-scratch-ts",
+    name: TemplateNames.ApiPluginFromScratch,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "api-plugin-from-scratch-bearer-js",
+    name: TemplateNames.ApiPluginFromScratchBearer,
+    language: "javascript",
+    description: "",
+  },
+  {
+    id: "api-plugin-from-scratch-bearer-ts",
+    name: TemplateNames.ApiPluginFromScratchBearer,
+    language: "typescript",
+    description: "",
+  },
+  {
+    id: "api-plugin-from-scratch-oauth-js",
```

Comment:
I feel putting such templates here will lead to use the default generator. You might want to check the content in packages\fx-core\src\component\generator\generatorProvider.ts

Reply:
or the order inside generatorProvider.ts need to be updated.

Reply:
List of templates which can be handled by default generator is updated to fix this overlapping.